### ISSUE TITLE
Add option to make adding the Authorization header optional

### ISSUE THREAD

**Issue by irealworlds** [auth0/auth0-angular#140]
**Please do not report security vulnerabilities here**. The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.

**Thank you in advance for helping us to improve this library!** Your attention to detail here is greatly appreciated and will help us respond as quickly as possible. For general support or usage questions, use the [Auth0 Community](https://community.auth0.com/) or [Auth0 Support](https://support.auth0.com/). Finally, to avoid duplicates, please search existing Issues before submitting one here.

By submitting an Issue to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

Currently, the default `HttpInterceptor` always throws an exception when trying to send a request to an uri in the configured `allowedList`. I think it would be useful in some cases to be able to configure it so that, for specific endpoints, the interceptor tries to add the Authorization header if possible (i.e. if the user is logged in), but still sends the requests anyway, even if that's not the case.

### Describe the ideal solution

Entries in the `allowedList` could have an aditional field, something like `continueOnError` (that defaults to `false`) that essentially means "I would **like** this to have the header, but I **need** it to be sent in any case.".

## Alternatives and current work-arounds

You can create a custom interceptor that always *tries* to add the header, but just leaves things untouched if that's not possible.

### Additional context

An example use case is an `/available-features` endpoint, that should return what features should be enabled for a particular user. This should not be inaccessible to requests that are not authorized, but authorization changes the result of requests to this endpoint, as permissions play a role in determining the status of a particular feature for the current user.

**Comment by stevehobbsdev** [comment]
> Currently, the default HttpInterceptor always throws an exception when trying to send a request to an uri in the configured allowedList

This sounds really odd; can you tell me more about the exception you're seeing?

**Comment by irealworlds** [comment]
> This sounds really odd; can you tell me more about the exception you're seeing?

From what I've seen, the `this.authService.getAccessTokenSilently(options)` part of the interceptor returns an error when the user is not logged in.

![image](https://user-images.githubusercontent.com/28299559/113717898-c074d180-96f4-11eb-80eb-9fa09cda4090.png)

Which, I get it, is actually the expected behaviour, as getting the token isn't something a "guest" user should do. What I'm proposing is adding the option to make the interceptor ignore this error on select endpoints.

**Comment by stevehobbsdev** [comment]
Right, that makes sense.

We talked about this internally and in general we think it's a good addition. We try not to create interceptors that support very specific scenarios and instead encourage developers to forge their own to suit their needs, but we think in this case a flag that allows the interceptor to ignore this and continue anyway without a token would be a useful addition.

I've added it to our internal backlog for triage.
